packages/js/src/utils/find-npm-dependencies.ts

```diff
@@ -101,12 +104,20 @@ function collectDependenciesFromFileMap(
     inputs
   );
 
+  const excludedFilesRegex = productionOnly
+    ? // For prod-only deps, exclude known build config files as these are not used in prod runtime. e.g. vite.config.ts.
+      new RegExp(
+        joinPathFragments(
+          sourceProject.data.root,
+          // If users customized their names such that they don't match this pattern
+          // then they need to manually add "nx-ignore-next-line" to their build config files.
+          '(package.json|(esbuild|rollup|vite|webpack).config.[jt]s)'
```

Comment:
I'm wondering if instead of listing known files we should instead provide a field where the user would specify all the ignored files:
```JSON
{
  "@nx/dependency-checks": [
    "error",
    {
      "buildTargets": ["build", "custom-build"], // add non standard build target names
      "ignoredFiles" ["esbuild.config.js"]
    }
  ]
}
```

We could also support Regex like in the code base.

The default value can be:
```JSONC
[
	 "^(package.json|(esbuild|rollup|vite|webpack).config.[jt]s)$"
]
```